stdlib/string.mli

```diff
@@ -395,6 +395,48 @@ val is_valid_utf_16le : t -> bool
 (** [is_valid_utf_16le b] is [true] if and only if [b] contains valid
     UTF-16LE data. *)
 
+(** {1:spellchecking Spellchecking} *)
+
+val edit_distance : ?limit:int -> t -> t -> int
+(** [edit_distance s0 s1] is the number of single character edits
+    (understood as insertion, deletion, substitution, transposition)
+    that are needed to change [s0] into [s1].
+
+    If [limit] is provided the function returns with [limit] as soon
+    as it was determined that [s0] and [s1] have distance of at least
+    [limit]. This is faster if you have a fixed limit, for example for
+    spellchecking.
+
+    The function assumes the strings are UTF-8 encoded and uses {!Uchar.t}
+    for the notion of character. Decoding errors are replaced by
+    {!Uchar.rep}. Normalizing the strings to
+    {{:https://unicode.org/glossary/#normalization_form_c}NFC} gives
+    better results.
+
+    {b Note.} This implements the simpler optimal string alignement
```

Comment:
Nit, but I think adding a few capitals improves understanding of the comment (if you do not know what it is talking about beforehand).
```suggestion
    {b Note.} This implements the simpler Optimal String Alignement (OSA)
```